src/vecgeom/VecgeomTrackView.i.hh

```diff
@@ -144,6 +147,24 @@ CELER_FUNCTION real_type VecgeomTrackView::find_next_step()
     return next_step_;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Find the new safety at a given position.
```

Comment:
Does this work at any position in the geometry, or just inside the current volume? I assume the latter?

Reply:
You are right. It should work only inside the current volume.  Since checking whether the given position is inside the current volume or not is also expensive, the method should be used carefully if it is used outside the VecGeomTrackView, i.e, from MSC or potentially from the field propagator - the use in  move_internal(real_type dist) should be fine as the pos_ is always inside the current volume, but the input position for move_internal(const Real3& pos) should be checked and inside the current volume before it is called.  The current use case in UrbanMsc is to get the current safety (at the pre-step point, so either on the boundary or inside the current volume) in the default step limitation algorithm (or at the position and the direction with a given distance smaller than the next step, which is used in another step limitation algorithm).

Reply:
I think for all our use cases it should be fine, just like our "move internal" function we can't really verify it, but I just wanted to make sure I understood it and it's documented.